File: kata-7-kyu/dont-give-me-five/python/dont_give_me_five_test.py
# + =====================================================================================================================
# +
# + 7 kyu - Don't give me five!  [ ID: 5813d19765d81c592200001a ] (dont-give-me-five)
# + URL: https://www.codewars.com/kata/5813d19765d81c592200001a
# + Category: ALGORITHMS  |  Tags: MATHEMATICS | ARRAYS | ALGORITHMS
# +
# + =====================================================================================================================

import codewars_test as test
from dont_give_me_five import dont_give_me_five


@test.describe("Fixed Tests")
def fixed_tests():
    @test.it("Basic Test Cases")
    def basic_test_cases():
        test.assert_equals(dont_give_me_five(1, 9), 8)
        test.assert_equals(dont_give_me_five(4, 17), 12)

    @test.it("More Tests")
    def more_test_cases():
        test.assert_equals(dont_give_me_five(1, 90), 72)
        test.assert_equals(dont_give_me_five(-4, 17), 20)
        test.assert_equals(dont_give_me_five(-4, 37), 38)
        test.assert_equals(dont_give_me_five(-14, -1), 13)
        test.assert_equals(dont_give_me_five(-14, -6), 9)


@test.describe("Random Tests")
def _():

    from random import randint

    def dont_give_me_five_check(start, end):
        count = 0
        for i in range(start, end + 1):
            if "5" in list(str(i)):
                continue
            count += 1
        return count

    for h in range(1, 30):
        start = randint(50, 110)
        end = randint(start + 1, start + 80)
        result = dont_give_me_five_check(start, end)
        res = dont_give_me_five(start, end)

        @test.it(f"Testing for dont_give_me_five({start}, {end})")
        def _():
            test.assert_equals(res, result)

File: kata-7-kyu/dont-give-me-five/javascript/dontGiveMeFive.Test.js
//+ ====================================================================================================================
//+
//+ 7 kyu - Don't give me five!  [ ID: 5813d19765d81c592200001a ] (dont-give-me-five)
//+ URL: https://www.codewars.com/kata/5813d19765d81c592200001a
//+ Category: ALGORITHMS  |  Tags: MATHEMATICS | ARRAYS | ALGORITHMS
//+
//+ ====================================================================================================================

import { assert } from "vitest"
const { dontGiveMeFive } = require("./dontGiveMeFive")

describe("KataTests", function () {
  it("exampleTests", function () {
    assert.strictEqual(dontGiveMeFive(1, 9), 8)
    assert.strictEqual(dontGiveMeFive(4, 17), 12)
  })
  it("moreTests", function () {
    assert.strictEqual(dontGiveMeFive(1, 90), 72)
    assert.strictEqual(dontGiveMeFive(-4, 17), 20)
    assert.strictEqual(dontGiveMeFive(-4, 37), 38)
    assert.strictEqual(dontGiveMeFive(-14, -1), 13)
    assert.strictEqual(dontGiveMeFive(-14, -6), 9)
  })
  it("randomTests", function () {
    for (let r = 0; r < 30; r++) {
      let start = Math.floor(Math.random() * 110 - 50)
      let end = Math.floor(Math.random() * (80 - start - 1) + start + 1)

      let expectedCount = 0
      for (let i = start; i <= end; i++) {
        if (!(i + "").includes("5")) {
          expectedCount++
        }
      }

      assert.strictEqual(dontGiveMeFive(start, end), expectedCount)
    }
  })
})
